### PR TITLE
Core 1192 webpack bundles from memory

### PR DESCRIPTION
**Pull request by Aapzu**
Adds option to use the webpack bundles directly from the memory (webpack dev server). By default they're still written into real files, and used from there.
Some parameter names need to be thought a bit before merging, I'll comment them in the PR.
Please comment if something stays unclear (I'm not explaining more here since the README should do the trick)

### CODE REVIEW

File: grails-app/conf/Config.groovy [hunk 1]
```diff
@@ -69,7 +69,7 @@ environments {
 }
 
 // See WebpackTagLib.groovy
-webpack.bundle.dir = '/webpack-bundles'
+webpack.bundle.dir = System.getProperty("streamr.webpack.bundle.location") ?: '/webpack-bundles'
```
Comment (Aapzu) [inline grails-app/conf/Config.groovy]: Is `streamr.webpack.bundle.location` a good parameter name?

File: package.json [hunk 2]
```diff
@@ -15,11 +15,11 @@
...
     "test-old": "mocha web-app/js/unifina/test --recursive",
     "test-new": "node ./node_modules/.bin/jest ./web-app/react-app/tests",
-    "build": "NODE_ENV=production ./node_modules/.bin/webpack",
-    "build-dev": "./node_modules/.bin/webpack",
+    "build": "NODE_ENV=production webpack",
+    "build-dev": "webpack",
     "clean": "rm -rf node_modules/",
     "clean-install": "npm run clean && npm install",
-    "dev": "./node_modules/.bin/webpack-dev-server"
+    "dev": "webpack-dev-server --hot"
```
Comment (Aapzu) [inline package.json]: Hot reloading doesn't really work yet... But I'll try to get it work in the near future. In the meanwhile, adding the flag is not doing any harm.

File: package.json [hunk 3]
```diff
@@ -15,11 +15,11 @@
     "test": "npm run test-old; npm run test-new",
     "test-old": "mocha web-app/js/unifina/test --recursive",
     "test-new": "node ./node_modules/.bin/jest ./web-app/react-app/tests",
-    "build": "NODE_ENV=production ./node_modules/.bin/webpack",
-    "build-dev": "./node_modules/.bin/webpack",
+    "build": "NODE_ENV=production webpack",
```
Comment (Aapzu) [inline package.json]: When using `webpack` in npm script, it actually uses `./node_modules/.bin/webpack`.

File: README.md [hunk 4]
```diff
@@ -70,9 +70,20 @@ These are also available as pre-shared run configurations if you use IntelliJ ID
...
+The UI is increasingly implemented with JavaScript libraries React and Redux and compiled with Webpack. The source files must first be transpiled and compiled into bundle file(s). This happens by running `npm run build` in the root directory. Normally, when running the Grails app the build command is run automatically, and needs not to be cared of.
+
+#### Running the dev server
+
+When developing the UI, it's much more handy to run the development server in the background. This happens with `npm run dev`. 
+The dev server updates bundle files on-the-fly as changes are detected in source files.
+
+#### Extra
+
+Normally the bundles are always written into files, where Grails no to take them from. However, if you want to use the files straight from the memory of the Webpack dev server (for increased building speed), is that possible with the following commands:
```
Comment (harbu) [inline README.md]: `where Grails no to take them from` -> knows to take the from?

File: README.md [hunk 5]
```diff
@@ -70,9 +70,20 @@ These are also available as pre-shared run configurations if you use IntelliJ ID
...
+ 1) Run grails with `grails run-app -Dstreamr.webpack.bundle.location=http://localhost:9000` (9000 is the default port of the dev server)
```
Comment (harbu) [inline README.md]: We should have a shared IntelliJ runConfiguration for this and the `NO_FILES=true npm run dev`.

File: grails-app/conf/Config.groovy (hunk 1 above)
Comment (harbu) [reply grails-app/conf/Config.groovy]: Yeah good name. Should rename `webpack.bundle.dir` to `streamr.webpack.bundle.location` in code for consistency.

**Comment by Aapzu** [comment]
@harbu fixed

**Comment by harbu** [comment]
Looks good to me!

**Pull request merged**